src/internal/EnhancedSwitch.js

```diff
@@ -13,15 +13,15 @@ function getStyles(props, context) {
   return {
     root: {
       position: 'relative',
-      cursor: props.disabled ? 'default' : 'pointer',
+      cursor: props.disabled ? 'not-allowed' : 'pointer',
       overflow: 'visible',
       display: 'table',
       height: 'auto',
       width: '100%',
     },
     input: {
       position: 'absolute',
-      cursor: props.disabled ? 'default' : 'pointer',
+      cursor: props.disabled ? 'not-allowed' : 'pointer',
```

Comment:
@nktssh Awesome, yes IMHO the `EnhancedButton` and `EnhancedSwitch` are the most questionable changes.